File: base/src/main/java/de/whitefrog/frogr/health/GraphHealthCheck.java
package de.whitefrog.frogr.health;

import com.codahale.metrics.health.HealthCheck;
import de.whitefrog.frogr.Service;

public class GraphHealthCheck extends HealthCheck {
  private Service service;
  
  public GraphHealthCheck(Service service) {
    this.service = service;
  }
  
  @Override
  protected Result check() {
    if(service.graph() == null) {
      return Result.unhealthy("The service is not connected to a graph");
    }
    return Result.healthy();
  }
}

File: base/src/main/java/de/whitefrog/frogr/health/GraphHealthCheck.kt
package de.whitefrog.frogr.health

import com.codahale.metrics.health.HealthCheck
import de.whitefrog.frogr.Service

class GraphHealthCheck(private val service: Service) : HealthCheck() {
  public override fun check(): HealthCheck.Result {
    return if (service.graph() == null) {
      HealthCheck.Result.unhealthy("The service is not connected to a graph")
    } else HealthCheck.Result.healthy()
  }
}
